src/Workspaces/MSBuildTest/Utilities/VisualStudioMSBuildInstalled.cs

```diff
@@ -2,63 +2,52 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using Microsoft.Build.Locator;
+using System.Threading;
+using System.Threading.Tasks;
 using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.MSBuild.UnitTests
 {
     internal class VisualStudioMSBuildInstalled : ExecutionCondition
     {
-#if NET472_OR_GREATER
-
-        private static readonly VisualStudioInstance? s_instance;
-        private readonly Version _minimumVersion;
+        private static readonly string? s_skipReason;
 
         static VisualStudioMSBuildInstalled()
         {
-            var latestInstalledInstance = (VisualStudioInstance?)null;
-            foreach (var visualStudioInstance in MSBuildLocator.QueryVisualStudioInstances())
+            if (!PlatformInformation.IsWindows)
             {
-                if (latestInstalledInstance == null || visualStudioInstance.Version > latestInstalledInstance.Version)
-                {
-                    latestInstalledInstance = visualStudioInstance;
-                }
+                s_skipReason = "Test is only supported on Windows since it looks for a Visual Studio install.";
             }
-
-            if (latestInstalledInstance != null)
+            else if (!IsVisualStudioMSBuildInstalled())
             {
-                MSBuildLocator.RegisterInstance(latestInstalledInstance);
-                s_instance = latestInstalledInstance;
+                s_skipReason = "No usable Visual Studio is installed.";
             }
         }
 
-#endif
-
-        public VisualStudioMSBuildInstalled()
-            : this(new Version(17, 0))
+        private static bool IsVisualStudioMSBuildInstalled()
         {
-        }
+            BuildHostProcessManager? buildHostProcessManager = null;
 
-        internal VisualStudioMSBuildInstalled(Version minimumVersion)
-        {
-#if NET472_OR_GREATER
-            _minimumVersion = minimumVersion;
-#endif
+            try
+            {
+                buildHostProcessManager = new BuildHostProcessManager();
+
+                var buildHost = buildHostProcessManager.GetBuildHostAsync(BuildHostProcessManager.BuildHostProcessKind.NetFramework, CancellationToken.None).Result;
```

Comment:
Unfortunately the only way to use MSBuildLocator right now to find a VS install is to launch a .NET Framework process if you're on .NET Core. I've been talking to the MSBuild team to lift this requirement.

Reply:
MSBuildLocator generally only allows you to discover MSBuilds that match your runtime so a .NET Core process can't enumerate .NET Framework MSBuilds or vice versa. Before this PR, the tests that tested .NET Framework style projects only ran in net472, and SDK style projects in .NET Core, since that's all that was compatible. Now that we can launch processes to do that for us, we want to run the tests for either TFM as long as there's an MSBuild we can use. Unfortunately due to the current MSBuildLocator behavior we can't do that yet.

I do plan to follow-up with the MSBuild locator owners to discuss adding this. I mentioned it to one of them and there were some concerns about runtime compatibility we'd have to work through (it means loading SDK discovery in .NET Framework processes, and COM in .NET Core processes). I would absolutely like to be able to use MSBuildLocator here though because launching processes here is a bit icky -- it means that a catastrophic bug (say a hung buildhost) can then hang test discovery due to how our conditional facts work.